src/main/java/com/amazon/ion/impl/macro/MacroCompiler.kt

```diff
@@ -85,10 +85,23 @@ internal class MacroCompiler(
 
             // Read the next parameter name
             val annotations = getTypeAnnotationSymbols()
-            confirm(annotations.isEmptyOr(Macro.ParameterEncoding.Tagged.ionTextName)) { "unsupported parameter encoding $annotations" }
+            val parameterEncoding = when (annotations.size) {
+                0 -> Macro.ParameterEncoding.Tagged
+                1 -> {
+                    val encodingText = annotations[0].text
+                    val encoding = Macro.ParameterEncoding.entries.singleOrNull { it.ionTextName == encodingText }
+                    if (encoding == null) {
+                        // TODO: Check for macro-shaped parameter encodings, and only if it's still null, we throw.
+                        throw IonException("unsupported parameter encoding $annotations")
+                    }
+                    encoding
+                }
+                2 -> TODO("Qualified references for macro-shaped parameters")
+                else -> throw IonException("unsupported parameter encoding $annotations")
+            }
```

Comment:
This patch is contributed by @popematt, and is exercised by the added `taglessExpressionGroup` test.